projects/plugins/jetpack/_inc/client/recommendations/sidebar/mobile-app/apps-badge.jsx

```diff
@@ -0,0 +1,168 @@
+/**
+ * External dependencies
+ */
+
+import PropTypes from 'prop-types';
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import { startsWith } from 'lodash';
+import { translate } from 'i18n-calypso';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import { getUserLocale as getLocaleSlug } from 'components/number-format';
+import { imagePath } from 'constants/urls';
+import analytics from 'lib/analytics';
+
+/**
+ * Style dependencies
+ */
+import './apps-badge.scss';
+
+// the locale slugs for each stores' image paths follow different rules
+// therefore we have to perform some trickery in getLocaleSlug()
+const APP_STORE_BADGE_URLS = {
+	ios: {
+		defaultSrc: imagePath + '/get-apps-ios-store.svg',
+		src: 'https://linkmaker.itunes.apple.com/assets/shared/badges/{localeSlug}/appstore-lrg.svg',
+		tracksEvent: 'calypso_app_download_ios_click',
+		getStoreLink: utm_source =>
+			`https://apps.apple.com/app/apple-store/id335703880?pt=299112&ct=${ utm_source }&mt=8`,
+		getTitleText: () => translate( 'Download the WordPress iOS mobile app.' ),
+		getAltText: () => translate( 'Apple App Store download badge' ),
+		getLocaleSlug: function () {
+			const localeSlug = getLocaleSlug();
+			const localeSlugPrefix = localeSlug.split( '-' )[ 0 ];
+			return localeSlugPrefix === 'en' ? 'en-us' : `${ localeSlugPrefix }-${ localeSlugPrefix }`;
+		},
+	},
+	android: {
+		defaultSrc: imagePath + '/get-apps-google-play.png',
+		src:
+			'https://play.google.com/intl/en_us/badges/images/generic/{localeSlug}_badge_web_generic.png',
+		tracksEvent: 'calypso_app_download_android_click',
```

Comment:
Here I've just taken the component from Calypso and modified it slightly to work within Jetpack, so this is just me taking the behavior there.

In the event that loading the badge fails this component will default to using the local badge. For both iOS and Android the locale is in the URL. So if either the URL changes or if the language is not supported it will end up using the default English ones that are available locally.